crawl-ref/source/mon-place.cc

```diff
@@ -391,26 +324,22 @@ void spawn_random_monsters()
     if (!x_chance_in_y(5, rate))
         return;
 
-    // Place normal dungeon monsters, but not in player LOS. Don't generate orb
-    // spawns in Abyss to show some mercy to players that get banished there on
-    // the orb run.
-    if (player_in_connected_branch()
-        || (player_on_orb_run() && !player_in_branch(BRANCH_ABYSS)))
+    // Orb spawns. Don't generate orb spawns in Abyss to show some mercy to
+    // players that get banished there on the orb run.
+    if (player_on_orb_run() && !player_in_branch(BRANCH_ABYSS))
```

Comment:
I don't think you should change it in this PR, but seeing this again made me think: is this really a necessary mercy? If I ever get banished during the orb run it's been a big relief, a nice chance to heal up. Maybe we should just let the orb run keep chasing you.

Reply:
Have you been banished on the orbrun since banishment depth is variable?  At that point in the game you will almost certainly get banished to abyss 5, which isn't much of a break from the orbrun in my experience, especially on a non-stealthy character.  IIRC when this has happened to me the only viable way out was by fighting enough monsters to open a portal.

Reply:
@rawlins I've actually just gotten the rune when this deep banishment happens on the orb run, since that's easy to do (esp without orb spawns), and it makes exits appear frequently, so I don't think killing monsters is the only way. Killing monsters may well be a faster way provided nothing too horrible appears.

Having orb spawns in Abyss would arguably be a nerf to Lucy's self-banishment, but I don't feel strongly either way. @amalloy If you find that there's agreement that this orb spawn protection is unecessary, feel free to remove it, but as you say I'll keep it in this PR.